### DISCUSSION TITLE
[A/L] (2021/22) Funzioni - Esercizio 22

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#262] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152316929-eb4e77a0-fbfa-47fb-8b15-3019ab0da619.png)

**Comment by LuginiAndrea** [comment]
La prima è vera, interpretiamo la suriettiva di $f$ in un caso particolare
$$\exists x \in \mathbf{N} \textrm{ t.c. }f\left(x\right) = 2$$
Di conseguenza $g$, non essendo definita in $2$, non si può avverare la seguenta proposizione, necessaria affinchè $g$ possa essere considerata l'inversa di $f$
$$\left(g \circ f\right)\left(2\right) = 2$$

La seconda invece è vera:
Consideriamo $f\left(x\right) = 3x$ e $g\left(x\right) = \frac{x}{3}$. $\forall n \in \mathbf{N}, f\left(n\right) \neq 2$, di conseguenza $g$ è sempre definita nell'immagine di $f$ e quindi di effettuare l'operazione inversa di $3x$, $\frac{x}{3}$

Sinceramente la terza mi pare troppo semplice data la mia intepretazione, quindi credo si intenda un'altra cosa rispetto a quello che penso.
Detto ciò, la richiesta è che, data $g\left(x\right) = x$, $\left(g \circ f\right)\left(x\right) = x$.
Basti considerare una qualunque funzione $f$ tale che $2 \notin \textrm{Immagine}_f$ e per il quale vale
$\forall x \in \mathbf{N}, f\left(x\right) \neq x$.
Dimostriamo con un esempio per maggiore chiarezza: riprendiamo $f\left(x\right) = 3x$, vuol dire che 
$\left(g \circ f\right)\left(x\right) = g\left(3x\right) = 3x \neq x \Longrightarrow g \circ f \neq$ Identità
